mmcls/models/backbones/mlp_mixer.py

```diff
@@ -0,0 +1,218 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import torch.nn as nn
+from mmcv.cnn import build_norm_layer
+from mmcv.cnn.bricks.transformer import FFN
+from mmcv.runner.base_module import BaseModule, ModuleList
+
+from ..builder import BACKBONES
+from ..utils import PatchEmbed, to_2tuple
+from .base_backbone import BaseBackbone
+
+
+class MixerBlock(BaseModule):
+    """Implements mixer block in MLP Mixer.
+
+    Args:
+        num_tokens (int): The number of patched tokens
+        embed_dims (int): The feature dimension
+        tokens_mlp_dims (int): The hidden dimension for tokens FFNs
+        channels_mlp_dims (int): The hidden dimension for channels FFNs
+        drop_rate (float): Probability of an element to be zeroed
+            after the feed forward layer. Defaults to 0.
+        drop_path_rate (float): Stochastic depth rate. Defaults to 0.
+        num_fcs (int): The number of fully-connected layers for FFNs.
+            Defaults to 2.
+        act_cfg (dict): The activation config for FFNs.
+            Defaluts to ``dict(type='GELU')``.
+        norm_cfg (dict): Config dict for normalization layer.
+            Defaults to ``dict(type='LN')``.
+        init_cfg (dict, optional): Initialization config dict.
+            Defaults to None.
+    """
+
+    def __init__(self,
+                 num_tokens,
+                 embed_dims,
+                 tokens_mlp_dims,
+                 channels_mlp_dims,
+                 drop_rate=0.,
+                 drop_path_rate=0.,
+                 num_fcs=2,
+                 act_cfg=dict(type='GELU'),
+                 norm_cfg=dict(type='LN'),
+                 init_cfg=None):
+        super(MixerBlock, self).__init__(init_cfg=init_cfg)
+
+        self.norm1_name, norm1 = build_norm_layer(
+            norm_cfg, embed_dims, postfix=1)
+        self.add_module(self.norm1_name, norm1)
+        self.token_mix = FFN(
+            embed_dims=num_tokens,
+            feedforward_channels=tokens_mlp_dims,
+            num_fcs=num_fcs,
+            ffn_drop=drop_rate,
+            dropout_layer=dict(type='DropPath', drop_prob=drop_path_rate),
+            act_cfg=act_cfg)
+
+        self.norm2_name, norm2 = build_norm_layer(
+            norm_cfg, embed_dims, postfix=2)
+        self.add_module(self.norm2_name, norm2)
+        self.channel_mix = FFN(
+            embed_dims=embed_dims,
+            feedforward_channels=channels_mlp_dims,
+            num_fcs=num_fcs,
+            ffn_drop=drop_rate,
+            dropout_layer=dict(type='DropPath', drop_prob=drop_path_rate),
+            act_cfg=act_cfg)
+
+    @property
+    def norm1(self):
```

Comment:
Since we use `norm_cfg` to build all kinds of norm layers, the `build_norm_layer` function will infer the abbreviation of these layers.
For example, `LayerNorm` -> `ln`, `BatchNorm` -> `bn`. It's helpful to set proper layer keys in checkpoints. So it's acceptable to use this naming format. 
Of course, using `norm1` & `norm2` directly is also acceptable. There is no strict naming convention.